adapter/oidc/src/main/kotlin/com/xorker/xpr/oidc/apple/AppleOidcHandler.kt

```diff
@@ -0,0 +1,36 @@
+package com.xorker.xpr.oidc.apple
+
+import com.xorker.xpr.exception.UnAuthenticationException
+import com.xorker.xpr.oidc.apple.dto.ApplePublicKeys
+import feign.FeignException
+import org.springframework.stereotype.Component
+
+@Component
+internal class AppleOidcHandler(
+    private val httpClient: AppleFeignClient,
+    private val parser: AppleIdTokenParser,
+    private val matcher: ApplePublicKeyMatcher,
+    private val keyGenerator: PublicKeyGenerator,
+    private val validator: AppleIdTokenValidator,
+) {
+    internal fun getPlatformUserId(token: String): String {
+        val applePublicKeys = getApplePublicKeys()
+
+        val header = parser.parseHeader(token)
+        val applePublicKey = matcher.getMatchedApplePublicKey(token, applePublicKeys.keys, header)
+
+        val publicKey = keyGenerator.generatePublicKey(applePublicKey)
+
+        val claims = parser.getClaims(token, publicKey)
+        validator.validateClaims(claims)
+        return claims.subject
```

Comment:
p2. 헨들러가 굳이 알 필요 없는 저수준의 로직까지 함께 포함하고 있는 것 같아요.

저는 Handler가 굳이 JWT 구조를 알 필요가 있나 생각이 들어요.

그리고 getClaims 역시 단순히 Claim을 가져오는 것처럼 보이네요. (특히 위에서 똑같은 parser을 이용해서 Header도 그냥 가져오다보니, 뭔가 Exception을 던질 것 처럼 보이지가 않아요.)
Parser의 역할이 커진 것 같습니다. 전부 Validator에 합쳐도 괜찮아 보여요.

```suggestion
        val publicKey = applePublicKeyService.generatePublicKey(token)
        
        return validator.validateAndGetSubject(token, publicKey)
```

generatePublicKey 해야 할 일
1. publicKey 목록 가져오기
2. token과 일치하는 publicKey 찾기
3. publicKey 발급

validateAndGetSubject 해야 할 일
1. Jwt 검증
2. Apple 관련 추가 검증
3. subject 반환